yarn-project/foundation/src/serialize/types.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Represents a fixed-length array.
+ */
+export type Tuple<T, N extends number> = N extends N ? (number extends N ? T[] : _Tuple<T, N, []>) : never;
+/**
+ * Recursive type helper for constructing a fixed-length tuple of a given type.
+ * This is utilized internally by Tuple to create the final fixed-length tuple.
+ */
+type _Tuple<T, N extends number, R extends unknown[]> = R['length'] extends N ? R : _Tuple<T, N, [T, ...R]>;
+
+/**
+ * Check an array size, and cast it to a tuple.
+ * @param array - The array.
+ * @param n - The size.
+ * @returns The case tuple, or throws Error.
+ */
+export function assertTuple<T, N extends number>(array: T[], n: N): Tuple<T, N> {
+  if (array.length !== n) {
+    throw new Error("Wrong 'fixed array' size");
+  }
+  return array as Tuple<T, N>;
+}
```

Comment:
Consider rename as it is unclear what this function does from its name.

Reply:
Will do

Reply:
Made it assertLength, kicked the existing assertLength function to be named assertMemberLength